### PR TITLE
#869 cleanups and hardening of tests around date-related types

### PR DESCRIPTION
**Pull request by Ocramius**
See #869

### CODE REVIEW

File: lib/Doctrine/DBAL/Types/ConversionException.php [hunk 1]
```diff
@@ -69,25 +69,31 @@ static public function conversionFailedFormat($value, $toType, $expectedFormat)
...
+        $actualType = is_object($value) ? get_class($value) : gettype($value);
+
+        if (is_scalar($value)) {
+            return new self(sprintf(
+                "Could not convert PHP value '%s' of type '%s' to type '%s'. Expected one of the following types: %s",
+                $value,
+                $actualType,
+                $toType,
+                implode(', ', $possibleTypes)
+            ));
```
Comment (zeroedin-bill) [inline lib/Doctrine/DBAL/Types/ConversionException.php]: This makes vastly more sense than the nonsense I wrote.

**Comment by zeroedin-bill** [comment]
The recently added DateTimeInterval type needs the same treatment. I can take care of that if needed. This is ready to merge.

File: tests/Doctrine/Tests/DBAL/Types/DateTest.php [hunk 2]
```diff
@@ -3,64 +3,31 @@
 namespace Doctrine\Tests\DBAL\Types;
 
 use Doctrine\DBAL\Types\Type;
-use Doctrine\Tests\DBAL\Mocks\MockPlatform;
 
-class DateTest extends \Doctrine\Tests\DbalTestCase
+class DateTestCase extends BaseDateTypeTestCase
```
Comment (zeroedin-bill) [inline tests/Doctrine/Tests/DBAL/Types/DateTest.php]: Doesn't matter if nobody extends this (probably nobody will) but the class name doesn't match the filename anymore.

File: tests/Doctrine/Tests/DBAL/Types/DateTimeTest.php [hunk 3]
```diff
@@ -2,54 +2,34 @@
 
 namespace Doctrine\Tests\DBAL\Types;
 
-use Doctrine\DBAL\Types\ConversionException;
 use Doctrine\DBAL\Types\Type;
-use Doctrine\Tests\DBAL\Mocks\MockPlatform;
 
-class DateTimeTest extends \Doctrine\Tests\DbalTestCase
+class DateTimeTestCase extends BaseDateTypeTestCase
```
Comment (zeroedin-bill) [inline tests/Doctrine/Tests/DBAL/Types/DateTimeTest.php]: class name != filename

File: tests/Doctrine/Tests/DBAL/Types/DateTimeTzTest.php [hunk 4]
```diff
@@ -3,53 +3,33 @@
 namespace Doctrine\Tests\DBAL\Types;
 
 use Doctrine\DBAL\Types\Type;
-use Doctrine\Tests\DBAL\Mocks\MockPlatform;
 
-class DateTimeTzTest extends \Doctrine\Tests\DbalTestCase
+class DateTimeTzTestCase extends BaseDateTypeTestCase
```
Comment (zeroedin-bill) [inline tests/Doctrine/Tests/DBAL/Types/DateTimeTzTest.php]: class name != filename

File: tests/Doctrine/Tests/DBAL/Types/TimeTest.php [hunk 5]
```diff
@@ -3,56 +3,28 @@
 namespace Doctrine\Tests\DBAL\Types;
 
 use Doctrine\DBAL\Types\Type;
-use Doctrine\Tests\DBAL\Mocks\MockPlatform;
 
-class TimeTest extends \Doctrine\Tests\DbalTestCase
+class TimeTestCase extends BaseDateTypeTestCase
```
Comment (zeroedin-bill) [inline tests/Doctrine/Tests/DBAL/Types/TimeTest.php]: class name != filename

File: tests/Doctrine/Tests/DBAL/Types/DateTest.php (hunk 2 above)
Comment (Ocramius) [inline tests/Doctrine/Tests/DBAL/Types/DateTest.php]: Urgh, silly IDE did silly refactoring. Fixing.

File: tests/Doctrine/Tests/DBAL/Types/DateIntervalTest.php [hunk 6]
```diff
@@ -7,14 +7,23 @@
...
 
+    /**
+     * {@inheritDoc}
+     */
     protected function setUp()
     {
-        $this->_platform = new MockPlatform();
-        $this->_type = Type::getType('dateinterval');
+        $this->platform = new MockPlatform();
+        $this->type     = Type::getType('dateinterval');
```
Comment (zeroedin-bill) [inline tests/Doctrine/Tests/DBAL/Types/DateIntervalTest.php]: Assert $this->type is instance of Type

**Comment by zeroedin-bill** [comment]
Big :+1: Thanks!

File: tests/Doctrine/Tests/DBAL/Types/DateIntervalTest.php (hunk 6 above)
Comment (Ocramius) [inline tests/Doctrine/Tests/DBAL/Types/DateIntervalTest.php]: done at Ocramius/dbal@155f5d754390ec7fabcf1a30015ffbbb2e096c86

**Pull request merged**